network/manager.go

```diff
@@ -0,0 +1,569 @@
+		modified = true
+	}
+	if modified {
+		return m.saveNetworkData()
+	}
+
+	return nil
+}
+
+func (m *Manager) AuthorizeMemeber(n *nodeInfo) error {
+	m.Lock()
+	defer m.Unlock()
+
+	if nw, ok := m.networkData.Waiting[n.Address]; ok {
+		ip, err := m.GetAvailableIp()
+		if err != nil {
+			log.Println("GetAvailableIp error:", err)
+			return err
+		}
+		nw.IP = ip
+		nw.Netmask = m.networkData.Netmask
+		m.networkData.Member[n.Address] = nw
+
+		if m.opts.Verbose {
+			log.Printf("AuthorizeMemeber, Member: %+v\n", m.networkData.Member)
+		}
+
+		delete(m.networkData.Waiting, n.Address)
+
+		if err = m.saveNetworkData(); err != nil {
+			return err
+		}
+
+		noti := &managerToMember{
+			MsgType:     NOTI_AUTHORIZED,
+			NetworkInfo: m.networkData.NetworkInfo,
+			NodeInfo:    []*nodeInfo{nw},
+		}
+
+		if _, err = SendMsg(m.c, n.Address, noti, false); err != nil {
+			log.Printf("Send NOTI_AUTHORIZED to %v error %v\n", n.Address, err)
+		}
+
+		for _, n := range m.networkData.Member {
+			acceptAddresses := m.networkData.AcceptAddress[n.Address]
+			if len(acceptAddresses) > 0 && acceptAddresses[0] == AcceptAll {
+				noti := &managerToMember{
+					MsgType: NOTI_SET_ACCEPT,
+				}
+				if _, err = SendMsg(m.c, n.Address, noti, false); err != nil {
+					log.Printf("Send NOTI_SET_ACCEPT to %v error %v\n", n.Address, err)
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func (m *Manager) DeleteWaiting(n *nodeInfo) error {
+	m.Lock()
+	defer m.Unlock()
+	delete(m.networkData.Waiting, n.Address)
+
+	return m.saveNetworkData()
+}
+
+func (m *Manager) UnauthorizeMemeber(n *nodeInfo) error {
+	m.Lock()
+	defer m.Unlock()
+
+	if nw, ok := m.networkData.Member[n.Address]; ok {
+		m.networkData.Waiting[n.Address] = nw
+		delete(m.networkData.Member, n.Address)
+
+		err := m.saveNetworkData()
+		if err != nil {
+			log.Println("saveNetworkData error:", err)
+			return err
+		}
+
+		for _, node := range m.networkData.Member {
+			changed := false
+			acceptAddresses := m.networkData.AcceptAddress[node.Address]
+			if len(acceptAddresses) > 0 && acceptAddresses[0] == AcceptAll {
+				changed = true
+			} else {
+				for index, addr := range acceptAddresses {
+					if addr == n.Address {
+						acceptAddresses = append(acceptAddresses[:index], acceptAddresses[index+1:]...)
+						m.networkData.AcceptAddress[node.Address] = acceptAddresses
+						changed = true
+						break
+					}
+				}
+			}
+			if changed {
+				noti := &managerToMember{
+					MsgType: NOTI_SET_ACCEPT,
+				}
+				if _, err = SendMsg(m.c, node.Address, noti, false); err != nil {
+					log.Printf("Send NOTI_SET_ACCEPT to %v error %v\n", node.Address, err)
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func (m *Manager) UpdateName(address, name string) error {
+	m.Lock()
+	defer m.Unlock()
+
+	if name == "" {
+		return nil
+	}
+
+	if addr, ok := m.networkData.NameToAddress[name]; ok {
+		if addr == address {
+			return nil
+		}
+		return errNameExist
+	}
+
+	var ok1, ok2 bool

```

Comment:
ok1 is defined twice.
Also I would recommend using `isMember` and `isWaiting`

Reply:
Yes, modified.